### ISSUE TITLE
Make the class WebDidUrlResolver public

### ISSUE THREAD

**Issue by YassirSellami** [eclipse-edc/Connector#3170]
# Feature Request

Is there any reason why the class: `org.eclipse.edc.iam.did.web.resolution.WebDidUrlResolver` is not public ?

## Which Areas Would Be Affected?
Did Web extension

## Why Is the Feature Desired?
This should help if one would prefer to create or use another WebDidResolver without rewriting that part of code

## Solution Proposal
`public class WebDidUrlResolver...`

**Comment by YassirSellami** [comment]
> `WebDidUrlResolver` is not an EDC extension point and is an internal implementation detail of that module. What is the use-case for extending it?

I need to add an `AtomicConstraintFunction` that retrieves a did in a raw format not a `DidDocument` class to be able to verify the integrity of that did against a Verifiable Credential issued by GAIA-X for Self-Description Compliance.

**Comment by jimmarino** [comment]
You should either use the SPI extension point `DidResolverRegistry` to resolve the DID or create your own if your function requires a custom format.

**Comment by YassirSellami** [comment]
Indeed I already created my own while using the same code of that class, that's why I though it would reduce the duplication. However,  isn't there a possibility to create a new method in the  DidResolver that simply returns the body in String or ResponseBody so the mapping would be done afterwards in those cases?

> You should either use the SPI extension point `DidResolverRegistry` to resolve the DID or create your own if your function requires a custom format.

**Comment by jimmarino** [comment]
That's not something we would want to support in an extension point. Just use OkHttp to resolve the DID. I'm also not sure why this operation would be performed in a policy function - it seems like the wrong place to do this. Policy functions are intended to evaluate the contents of a claim, not resolve or validate it. The latter should be done by the `IdentityService` or one of its extension delegates.

**Comment by YassirSellami** [comment]
> That's not something we would want to support in an extension point. Just use OkHttp to resolve the DID. I'm also not sure why this operation would be performed in a policy function - it seems like the wrong place to do this. Policy functions are intended to evaluate the contents of a claim, not resolve or validate it. The latter should be done by the `IdentityService` or one of its extension delegates.

Great thanks for your point, I will look into the IdentityService more in detail. The goal is to verify if the claim is still valid at the time of the negotiation.

**Comment by jimmarino** [comment]
> > That's not something we would want to support in an extension point. Just use OkHttp to resolve the DID. I'm also not sure why this operation would be performed in a policy function - it seems like the wrong place to do this. Policy functions are intended to evaluate the contents of a claim, not resolve or validate it. The latter should be done by the `IdentityService` or one of its extension delegates.
> 
> Great thanks for your point, I will look into the IdentityService more in detail. The goal is to verify if the claim is still valid at the time of the negotiation.

This should be done by the `IdentityService` (or a delegate extension) and placed in the `ParticipantAgent` context. There may be some work we need to do to support VP verification more generally.

**Comment by bscholtes1A** [comment]
> Hello @jimmarino thanks again for the feedback, I discussed it with @bscholtes1A and indeed it would better for the verification part to be done by the `CredentialVerifier`. However, there is still the issue with the resolver, I think it would be beneficial to have a DidResolver which returns a JSON-LD which is more and more used with the new protocol to my understanding, and which could be helpful to resolve verifiable credentials in a more generic format, without the need to create a custom implementation each time. Do you think that could be implemented in the identity-did-web extension or is it out of scope ?

@jimmarino basically, as part of the verification process of the Gaia-X VC, a `did` contained in the claims of this VC is resolved (for now this `did` is always a `did:web` but anyway). The issue is that this `did` does not resolve to a `DidDocument` but to a Gaia-X Self-Description. 

Thus the question of @YassirSellami is: is there any way we can make the DidResolver of the EDC something more generic? (maybe the output type can be a template?), so that the `did` resolution logic can be reused even if in cases where we do not want to resolve directly to a Did Document?

**Comment by YassirSellami** [comment]
> A few comments and a question:
> 
> 1. `DidResolverRegistry` should be used. We should not expose `WebDidResolver` as that is an implementation and not something that should be exposed through the public SPI. Using the registry will adhere to the SPI and make the code more general, since other resolution methods can be supported.
> 2. Does the GAIA-X self-description adhere to [The W3C DID Specification](https://www.w3.org/TR/did-core/)? If so, the `DidResolverRegistry` should return a representation that adheres to the W3C data model. If the self-description does not adhere to that model, then the resolution method is not Web DID and should not use the [did:web scheme](https://w3c-ccg.github.io/did-method-web/).
> 
> Assuming the GAIA-X self-description is contained in a properly structured DID document, we should follow the EDC approach to provide typed model classes that can be transformed from expanded Json-Ld. It looks as if the existing `DidDocument` can be refactored to support that as part of a broader overhaul of the Identity Hub.

I agree on the first point there is no need to change how the WebDidResolver already works with the registry, this could simply be a different implementation.
On the second point, no the Gaia-x Self description does not adhere to the DID specs such as a did document used as a verification method, however I believe it could still be used basically as an URI, to my understanding Verifiable Credentials can be referred to by a did:web.

**Comment by jimmarino** [comment]
> On the second point, no the Gaia-x Self description does not adhere to the DID specs such as a did document used as a verification method, however I believe it could still be used basically as an URI, to my understanding Verifiable Credentials can be referred to by a did:web.

If that is the case then the resolution method is not `did:web` but a custom, non-W3C compliant URN scheme. The DID resolvers should not be used (create a separate implementation). If `did:web` is used, a DID Document should be returned containing the VC or a URL should be used if returning a different format. Hijacking the `did:web` scheme like this is incompatible with W3C standards and will break implementations (as is shown with this issue).
